### PR TITLE
Allows carded AI's to be inserted into an AI upload

### LINKED ISSUE #27155: feature request: give AI uploads intellicard slots
#### Description of issue

Would be nice to be able to upload a lawset to (dead) carded AI - would finally allow for de-antagging them as well.

### PR DESCRIPTION
**Pull request by gurfan**
Closes #27155 

Carded AI's can now be inserted into an AI Upload. Slapping the upload with an loaded intellicard will place the AI within the upload (and leave you with an empty intellicard). Hitting the upload with an empty hand will set the uploaded AI for law changes. Nonfunctional (dead) AI's that have been carded can have their laws changed this way.

[tweak]

:cl:
 * rscadd: Carded AI's can now be inserted into an AI Upload for law changes. Dead AI's can now have their laws changed this way.

### CODE REVIEW

**Comment by SonixApache** [comment]
wow that was quick
nice job

File: code/game/machinery/computer/ai_core.dm [hunk 1]
```diff
@@ -254,6 +254,42 @@ That prevents a few funky behaviors.
...
+									to_chat(A, "You have been uploaded to a stationary terminal. Sadly, there is no remote access from here.")
+									to_chat(U, "<span class='notice'><b>Transfer successful</b>:</span> [A.name] ([rand(1000,9999)].exe) installed and executed successfully. Local copy has been removed.")
+						else
+							if(!C.contents.len && T.occupant)
+								C.name = "inteliCard - [T.occupant.name]"
+								if (T.occupant.stat == 2)
+									C.icon_state = "aicard-404"
+								else
+									C.icon_state = "aicard-full"
+								to_chat(T.occupant, "You have been downloaded to a mobile storage device. Still no remote access.")
```
Comment (Kurfursten) [inline code/game/machinery/computer/ai_core.dm]: It may have remote access. Check `T.occupant.control_disabled`

Comment (gurfan) [reply code/game/machinery/computer/ai_core.dm]: ive added a check that tells the AI if they have remote access upon being transferred back into the card

File: code/game/machinery/computer/law.dm [hunk 2]
```diff
@@ -6,10 +6,20 @@
...
 	var/opened = 0
 
 	light_color = "#555555"
 
+/obj/machinery/computer/aiupload/attackby(I as obj, user as mob)
+	if(istype(I, /obj/item/device/aicard))
+		if(stat & (NOPOWER|BROKEN))
+			to_chat(user, "This terminal isn't functioning right now, get it working!")
+			return
+		I:transfer_ai("AIUPLOAD","AICARD",src,user)
```
Comment (Kurfursten) [inline code/game/machinery/computer/law.dm]: no : please
you gotta cast this as an AI card, I know it adds a line but it's our coding standards

File: code/game/machinery/computer/law.dm [hunk 3]
```diff
@@ -39,7 +49,7 @@
 		to_chat(usr, "Law uploads have been disabled by Nanotrasen!")
 		return 0
 
-	if(current.stat == 2 || current.control_disabled == 1)
+	if(current.stat == 2 && src.occupant != current)
```
Comment (Kurfursten) [inline code/game/machinery/computer/law.dm]: src isn't necessary

File: code/game/machinery/computer/law.dm [hunk 4]
```diff
@@ -132,12 +142,18 @@
 		to_chat(usr, "The upload computer is broken!")
 		return
 
-	current = select_active_ai(user)
+	if (src.occupant)
+		current = src.occupant
```
Comment (Kurfursten) [inline code/game/machinery/computer/law.dm]: src

File: code/game/machinery/computer/law.dm [hunk 5]
```diff
@@ -132,12 +142,18 @@
...
+	else
+		current = select_active_ai(user)
 
 	if(!current)
 		to_chat(usr, "No active AIs detected.")
 	else
-		to_chat(usr, "[current.name] selected for law changes.")
+		if(src.occupant)
```
Comment (Kurfursten) [inline code/game/machinery/computer/law.dm:154]: src

**Pull request merged**